tests_e2e/orchestrator/lib/agent_test_suite.py

```diff
@@ -462,12 +462,19 @@ def _execute(self, environment: Environment, variables: Dict[str, Any]):
                                 test_suite_success = False
                                 raise
 
+                        # initialize and save test classes at once to avoid multiple initialization and also keep the context of the test class around
```

Comment:
The code would be simpler and clearer if you move the call to self._check_agent_log() into execute_test_suite().

As test are executed, we can collect the ignore rules into an array and then just pass them as argument to check_agent_log()

Reply:
let me work on it